File: modules/expo-bluesky-swiss-army/ios/Visibility/ExpoBlueskyVisibilityViewModule.swift
import ExpoModulesCore

public class Expokoyu.spaceVisibilityViewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("Expokoyu.spaceVisibilityView")

    AsyncFunction("updateActiveViewAsync") {
      VisibilityViewManager.shared.updateActiveView()
    }

    View(VisibilityView.self) {
      Events([
        "onChangeStatus"
      ])

      Prop("enabled") { (view: VisibilityView, prop: Bool) in
        view.enabled = prop
      }
    }
  }
}

File: modules/expo-bluesky-swiss-army/android/src/main/java/expo/modules/blueskyswissarmy/visibilityview/ExpoBlueskyVisibilityViewModule.kt
package expo.modules.blueskyswissarmy.visibilityview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class Expokoyu.spaceVisibilityViewModule : Module() {
  override fun definition() =
    ModuleDefinition {
      Name("Expokoyu.spaceVisibilityView")

      AsyncFunction("updateActiveViewAsync") {
        VisibilityViewManager.updateActiveView()
      }

      View(VisibilityView::class) {
        Events(arrayOf("onChangeStatus"))

        Prop("enabled") { view: VisibilityView, prop: Boolean ->
          view.isViewEnabled = prop
        }
      }
    }
}
